lib/wallet/account/hot-wallet-account.ts

```diff
@@ -34,7 +40,191 @@ export class HotWalletAccount implements IWallet {
+      }
+    }, 120000);
+
+    while (!eventReceived) {
+      await new Promise((resolve) => setTimeout(resolve, 500)); // wait and check every 500ms
+    }
+
+    clearTimeout(commitTimeout);
+
+    // Continue with reveal transaction after maturity event
+    eventReceived = false;
+    const { entries: newEntries } = await this.rpcClient.getUtxosByAddresses({
+      addresses: [address.toString()],
+    });
+
+    const revealUTXOs = await this.rpcClient.getUtxosByAddresses({
+      addresses: [P2SHAddress.toString()],
+    });
+
+    const { transactions: revealTransactions } = await createTransactions({
+      priorityEntries: [revealUTXOs.entries[0]],
+      entries: newEntries,
+      outputs: extraOutputs,
+      changeAddress: address.toString(),
+      priorityFee: revealPriorityFee,
+      networkId: this.networkId,
+    });
+    let revealHash: any;
+
+    for (const transaction of revealTransactions) {
+      console.log(`Reveal TX ID: ${transaction.id}`);
+      transaction.sign([privateKey], false);
+      const ourOutput = transaction.transaction.inputs.findIndex(
+        (input) => input.signatureScript === "",
+      );
+
+      if (ourOutput !== -1) {
+        const signature = transaction.createInputSignature(
+          ourOutput,
+          privateKey,
+        );
+        transaction.fillInput(
+          ourOutput,
+          script.encodePayToScriptHashSignatureScript(signature),
+        );
+      }
+      revealHash = await transaction.submit(this.rpcClient);
+      submittedTxId = revealHash;
+    }
+
+    const revealTimeout = setTimeout(() => {
+      if (!eventReceived) {
+        throw new Error("Reveal transaction did not mature within 2 minutes");
+      }
+    }, 120000);
+
+    while (!eventReceived) {
+      await new Promise((resolve) => setTimeout(resolve, 500)); // wait and check every 500ms
+    }
+
+    this.rpcClient.removeEventListener("utxos-changed", handleUtxosChanged);
+    clearTimeout(revealTimeout);
+    eventReceived = false;
+  }
+
+  async deploy(
+    payload: {
+      tick: string;
+      max: string;
+      lim: string;
+      dec: string;
+      pre: string;
+    },
+    extraOutputs: IPaymentOutput[] = [],
+  ): Promise<void> {
+    await this.performCommitReveal(
+      { p: "krc-20", op: "deploy", ...payload },
+      kaspaToSompi(Fee.Deploy.toString())!,
+      extraOutputs,
+    );
+  }
+
+  async mint(
+    payload: { tick: string },
+    extraOutputs: IPaymentOutput[] = [],
+  ): Promise<void> {
+    await this.performCommitReveal(
+      { p: "krc-20", op: "mint", ...payload },
+      kaspaToSompi(Fee.Mint.toString())!,
+      extraOutputs,
+    );
+  }
+
+  async transfer(payload: {
+    tick: string;
+    amt: string;
+    to: string;
+  }): Promise<void> {
+    await this.performCommitReveal(
+      { p: "krc-20", op: "transfer", ...payload },
+      kaspaToSompi(Fee.Base.toString())!,
+    );
+  }
```

Comment:
We can move these methods outside of wallet, it is the custom structure for KRC20.